### DISCUSSION TITLE
Undefined symbols for architecture arm64

### DISCUSSION THREAD

**Discussion by mihaicris-adoreme** [pointfreeco/swift-concurrency-extras#25] [Q&A]
### Description

I am getting an error in Xcode, when running tests, if I use swift-dependencies (which uses swift-concurrency-extras) at the linkage step

I am using OTHER_LDFLAGS -ld_classic

Do you support the classic linker? 

### Checklist

- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-concurrency-extras/issues) or [discussion](https://github.com/pointfreeco/swift-concurrency-extras/discussions).

### Expected behavior

Link project to succeed.

### Actual behavior

```
Undefined symbols for architecture arm64:
  "_$s17ConcurrencyExtras22withMainSerialExecutor9operationyyyKXE_tKF", referenced from:
      _$s16AdoreMeUnitTests019CheckoutAddressListD0C10invokeTestyyF in CheckoutAddressListTests.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### Steps to reproduce

```swift
final class CheckoutAddressListTests: XCTestCase {
    override func invokeTest() {
        withMainSerialExecutor {
            super.invokeTest()
        }
    }
```

### swift-concurrency-extras version information

1.1.0

### Destination operating system

iOS 17.4

### Xcode version information

15.3

### Swift Compiler version information

```shell
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
```

**Comment by stephencelis** [comment]
@mihaicris-adoreme Sorry, we must have missed this when it came in.

I don't think we do anything out of the ordinary here, except maybe `withMainSerialExecutor` relies on a Swift hook that is causing issues? That might make this a more general Swift bug with your linker settings and the hook we're using.

If you have a PR that addresses the issue we'd be open to merging it, or if you can at attach a project that reproduces the problem we'd take a look to see if we can figure out what's going on.

That said, I don't think we consider this a bug with the library given the less than traditional build settings, so I'm going to covert to a discussion for now.

**Reply by mihaicris-adoreme** [reply]
Issue no more reproduced.
